plutus-core/plutus-core/src/Crypto.hs

```diff
@@ -17,3 +28,25 @@ verifySignature pubKey msg sig =
             <$> publicKey pubKey
             <*> pure msg
             <*> signature sig
+
+verifySECP256k1Signature
+  :: BS.ByteString -- ^ Public key
+  -> BS.ByteString -- ^ Signature
+  -> BS.ByteString -- ^ Message hash
+  -> Emitter (EvaluationResult Bool)
+verifySECP256k1Signature pk sig msg =
+  case DSIGN.rawDeserialiseVerKeyDSIGN @SECP256k1DSIGN pk of
+    Nothing -> do
+      emitM "SECP256k1: Given invalid signing key."
+      pure EvaluationFailure
+    Just pk' -> case DSIGN.rawDeserialiseSigDSIGN @SECP256k1DSIGN sig of
+      Nothing -> do
+        emitM "SECP256k1: Given invalid signature."
+        pure EvaluationFailure
+      Just sig' -> case SECP.msg msg of
+        Nothing -> do
+          emitM "SECP256k1: Given invalid message hash."
+          pure EvaluationFailure
+        Just msg' -> case DSIGN.verifyDSIGN () pk' msg' sig' of
+          Left _   -> pure . pure $ False
+          Right () -> pure . pure $ True
```

Comment:
@kwxm bad news for ya, we have another emitting bulitin.

Reply:
Is this a problem in some sense I'm not aware of?

Reply:
@kozross, not a concern for you, just a pain in the ass to describe in the specification.

Reply:
I think we were really expecting this to be used only in the in the implementation of `trace` and not in arbitrary builtins, but I suppose it's harmless and it's commendable to give programmers extra feedback with something that it would be easy to get wrong.

I think from the point of specification, I'd inclined to just not mention this in the specification on the basis that the error messages are a detail specific to this particular implementation.  Trying to include the details of the logging mechanism in the specification would complicate things quite a bit, so maybe we could say that an implementation of Plutus Core may choose to provide facilities for logging errors as long as they don't have any effect on the semantics of built-in functions;  that's not really got anything to do with this PR though.

Off-topic, this makes me wonder if the logging mechanism could be used to do unexpected things in contract code. Could the off-chain part of a contract run some validation code (also off the chain), look at the log files and then branch on the contents to determine the future behaviour of the contract (which you could do even using `trace`)? That might provide a way to leak more information from validation than the `true`/`false` answer that we expect, but would that actually be a problem?